pyat/at/load/matfile.py

```diff
@@ -84,41 +85,79 @@ def _mat_encoder(v):
 def _matfile_generator(
     params: dict[str, Any], mat_file: str
 ) -> Generator[Element, None, None]:
-    """Run through Matlab cells and generate AT elements"""
+    """Run through Matlab cells and generate AT elements.
 
-    def mclean(data):
+    Arguments:
+        params: parameter dictionary
+        mat_file: matlab file name
+
+    Yields:
+        pyat Element from dictionary
+    """
+
+    def mclean(data: any) -> any:
         if data.dtype.type is np.str_:
             # Convert strings in arrays back to strings.
-            return str(data[0]) if data.size > 0 else ""
+            dataout = str(data[0]) if data.size > 0 else ""
         elif data.size == 1:
-            v = data[0, 0]
-            if issubclass(v.dtype.type, np.void):
+            vdata = data[0, 0]
+            if issubclass(vdata.dtype.type, np.void):
                 # Object => Return a dict
-                return {f: mclean(v[f]) for f in v.dtype.fields}
+                dataout = {f: mclean(vdata[f]) for f in vdata.dtype.fields}
             else:
                 # Return a scalar
-                return v
+                dataout = vdata
         else:
             # Remove any surplus dimensions in arrays.
-            return np.squeeze(data)
+            dataout = np.squeeze(data)
+        return dataout
+
+    def mcleanhdf5(data: any) -> any:
+        matlab_class = data.attrs["MATLAB_class"]
+        if matlab_class == b"char":
+            # Convert to string
+            dataout = "".join(chr(i) for i in np.asarray(data).flatten())
+        if matlab_class == b"double":
+            dataout = np.squeeze(np.asarray(data))
+        if matlab_class == b"struct":
+            dataout = {f: mcleanhdf5(data[f]) for f in data.keys()}
+        return dataout
```

Comment:
If for some custom attribute the Matlab class in not one of the expected ones, this crashes (`dataout` is undefined). It would be wise to add a default behaviour, returning `np.squeeze(np.asarray(data))`.